### PR TITLE
Made Files easier 3-11-18

### PR DESCRIPTION
**Pull request by lavaduderDev**
In Global_VM:

-  Added Force_Quit in project settings (You can now leave via the window's X button)

In Item.gd and scene_base.gd:
 

- added '.esc' to export(String,FILE) variables

Concept: 
 

- Tried making in game/editor Escoria syntax highlighter. For easier developer use.

### CODE REVIEW

**Comment by mjtorn** [comment]
On Sun, Mar 11, 2018 at 06:37:39AM +0000, lavaduderDev wrote:
>In Global_VM:
>
>-  Added Force_Quit in project settings (You can now leave via the window's X button)

I would change this to default to true.

>In Item.gd and scene_base.gd:
>
>- added '.esc' to export(String,FILE) variables

This narrows down the file types? This seems like a very welcome improvement,
I didn't even know that's possible!

>Concept:
>
>- Tried making in game/editor Escoria syntax highlighter. For easier developer use.

I may have to test this separately tomorrow, as I don't have anything like
code at hand now. Could you add some documentation on how to use it?

>-- File Changes --
>
>    M device/demo/rooms/test/chain_saw.tscn (23)
>    M device/demo/translations/general.en.translation (0)
>    M device/demo/translations/general.es.translation (0)
>    M device/demo/translations/general.fr.translation (0)
>    M device/globals/global_vm.gd (2)
>    M device/globals/global_vm.tscn (2)
>    M device/globals/item.gd (2)
>    M device/globals/scene_base.gd (2)
>    M device/globals/scene_main.tscn (2)
>    M device/project.godot (1)
>    A device/ui/escoria_writer.gd (36)
>    A device/ui/escoria_writer.tscn (163)

There are a great many things going on in this PR, touching many files.

My gut tells me it's all good and welcome work, but this isn't very
thematic at the moment. IMO splitting the commits make sense so people
can look at git annotate/blame and see why the change was made, but
they're all slight enough that one PR per commit is overkill.

Thanks a lot!

**Comment by lavaduderDev** [comment]
'Could you add some documentation on how to use it?'
Honestly it ain't usable, I was trying to make it so that when you clicked on an item in the editor, it'd show up. But right now it's not even saving properly. let alone ready to navigate the file system.

Your welcome.

**Comment by fleskesvor** [comment]
The work and pull request is much appreciated, @lavaduderDev. 🙂Like @mjtorn mentioned though, there's a lot going here, making it unsuitable for a single PR.

Some comments:

- The editor hint for .esc files is a great idea. I wasn't aware of the possibility, and this could easily be merged as a single PR.
- Configurable force quit is also a good idea, and can easily be merged as-is.
- There are two unused music files in the PR, demo_melody.mid and demo_melody.ogg
- The branching dialog with the NPC crashes the game because no dialogue resources are found in the resource loader. The issue is likely not introduced by your PR, since I don't see any changes on it in your commit, but until it's fixed, it can't be used in the demo scene.
- The changes on the old player character don't seem necessary. If you made the new sprites for the NPC and are fine with contributing them under the MIT license I'd be happy to include them once the issue above is fixed.
- The .esc script writer should be in a separate PR, since it's undocumented and experimental. I couldn't figure out how to use it, and it created several new files in my working directory.
- There are changes on inventory.tscn, inventory_items.tscn, dd_player.tscn, hud.tscn and the translation files, which I'm not sure what are, and maybe they aren't meant to be in this commit?

tl;dr: I think this should be tree separate PRs:

1. One for the editor hint and configurable force quit, as those can be merged as-is
2. One for the changes to the demo scene with the NPC and branching dialogue as it requires some fixing and discussion
3. One for the experimental script writer as it seems especially rough, and is a good topic for discussion and contributions by others

**Comment by lavaduderDev** [comment]
Forgive my incompetence. I have mostly used git for my own, not for a community project...sooo I have no idea what I am doing, or how to undo my crazy pull request. Should I delete my fork and start over?
'If you made the new sprites for the NPC and are fine with contributing them under the MIT license' -Everything I commit is MIT. Even the recently added music to showcase the broken MUSIC BG.

**Comment by fleskesvor** [comment]
No worries; mastering Git takes time and practice. 🙂

Don't delete anything. You just need to divide your work into a few branches.

The first thing you want to do is create a "backup" branch, since we're going to reset the branch tied to your pull request. To create a new branch named "backup", just run:

    git checkout -b backup

Of course, you can call this branch something else. Now, you might want to push this branch to your fork for safe-keeping, which you can typically just do with `git push` if your push strategy is set to matching, though if git complains about that, you can use `git push origin backup` instead.

Now switch back to your master branch and reset your branch to the last commit from upstream:

    git checkout master
    git reset 895a5337a2586b21057d74c05bf6fece9ab7032b

If you run `git status` now, you will see lots of unstaged files. We will stage some of them and add a few commits:

    git add device/globals/global_vm.gd
    git add device/project.godot
    git commit -m 'Make configurable force quit option'
    git add device/globals/item.gd
    git add device/globals/scene_base.gd
    git commit -m 'Add editor hint for .esc files'

You now have changes that diverges from your original graph, so to update origin/master, you'll have to use the `--force` option:

    git push --force

That will change the suggested changes in this pull request.

**Comment by lavaduderDev** [comment]
Okay, I'll avoid the demo add-ons, until the ?/dd_default.gd is fixed, The escoria writer will be made in a separate project, when ready, it'll be brought over.

**Comment by fleskesvor** [comment]
Ok. I will try to fix the branching dialogue in the meantime. Let me know if you want help with breaking your master branch up into new branches. I could make some pull requests on your fork.

**Comment by lavaduderDev** [comment]
So do want me to clean up code or something. Cause I'm in the dark about what is going on in my Pull Request. I have also made a new Pull request, that I didn't mean to add commits from this one to it, but it has it, so... Should I close this one, and let you just judge that sole pull request?

**Comment by fleskesvor** [comment]
No, no need to close this one. What happened in the other one is that you branched from the head of this branch instead of using a clean branch from upstream master.

If you just want to clean up this branch, you can follow these steps that I accidentally posted in the other PR first:

What you usually want to do is to rebase your commits on master of this repository, but since you currently have a PR on your master branch, you'd have to create a new branch for that.

These changes shouldn't be difficult to merge though, so don't worry about that now. We just need to clean up this branch, so make sure you have it checked out:

    git checkout escedit

Next, we remove all those extra commits by moving HEAD back to point at your second commit 

    git reset 9a48a8d8756a6dd9ea08686e73401f79bd46c5a0

Then all you have to do is push your changes again, but now you need to use `--force`, since we have to overwrite the remote branch

    git push --force origin escedit

If you need any help, feel free to ask here or on the `#escoria` channel on [Freenode IRC](https://webchat.freenode.net/).

**Comment by lavaduderDev** [comment]
I'm not going to bother with this PR anymore, too many commits, of various subjects, that to put it simply over complicated things, on various ends.

There are new pull requests, one dealing with subject matter of this one. This is isolated, and not in trouble city. End of this thing.

**Comment by fleskesvor** [comment]
Sorry for bailing on you on IRC last night. 🙁 I was writing from bed and fell asleep.

I see that you got the changes into new branches though, and that's great. 🙂The easiest way to get your master branch in sync with upstream now is to reset to the last commit from upstream and do a pull to get new changes.

    git reset 895a5337a2586b21057d74c05bf6fece9ab7032b
    git pull upstream master

If you also want to get rid of any changes on your branch, you can do a `git checkout .` to roll back changes on all tracked files and `git clean -f` to remove untracked files.
